api/ping.go

```
@@ -0,0 +1,81 @@
+// HandlerWebSocket handles WebSocket connections for the ping mechanism
+func (server *Server) HandlerWebSocket(c *gin.Context) {
+	// Upgrade HTTP connection to WebSocket
+	conn, err := upgrader.Upgrade(c.Writer, c.Request, nil)
+	if err != nil {
+		log.Printf("Failed to upgrade connection to WebSocket: %v", err)
+		return
+	}
+	defer conn.Close()
+
+	// Set read deadline to detect disconnections
+	conn.SetReadDeadline(time.Now().Add(60 * time.Second))
```

golangci-lint (1.64.8)

Error return value of `conn.SetReadDeadline` is not checked

(errcheck)